Spatial_Engine/Create/ShapeProfiles/TSectionProfile.cs

```diff
@@ -112,6 +112,11 @@ private static List<ICurve> TeeProfileCurves(double tft, double tfw, double wt,
             perimeter.Add(new Line { Start = p, End = p - xAxis * (tfw) });
             perimeter.Add(new Line { Start = origin + xAxis * (-wt / 2), End = origin + xAxis * (wt / 2) });
 
+            Point centroid = perimeter.IJoin().Centroid();
+            Vector tranlation = Point.Origin - centroid;
+
+            perimeter = perimeter.Select(x => x.ITranslate(tranlation)).ToList();
```

Comment:
You could return in this line already (same in other files)

Reply:
ah, true, will fix that